python/pyarrow/lib.pxd

```diff
@@ -142,6 +142,21 @@ cdef class PyExtensionType(ExtensionType):
     pass
 
 
+cdef class Metadata:
+    pass
```

Comment:
This is curious, is this needed

Reply:
I ended up removing collections.abc.Mapping because KeyValueMetadata is not a mapping, but implemented the mapping-like methods manually.

Reply:
Added it again for instance checks.